proof/crefine/X64/Finalise_C.thy

```diff
@@ -1987,7 +1985,7 @@ lemma ccap_relation_PDPT_IsMapped:
      capPDPTIsMapped_CL (cap_pdpt_cap_lift cap) = from_bool (\<not> Option.is_none x2)"
   apply (frule cap_get_tag_isCap_unfolded_H_cap)
   apply (clarsimp simp: ccap_relation_def cap_to_H_def cap_pdpt_cap_lift_def cap_lift_def Let_def
-                        cap_tag_defs to_bool_def true_def false_def
+                        cap_tag_defs to_bool_def true_def
```

Comment:
This is the final unfold that I wasn't able to remove. The goal is
```
¬ (cap_C.words_C cap.[0] >> 58) AND 1 = 0
    ⟹ (cap_C.words_C cap.[0] >> 58) AND 1 = signed true
```
Unfolding `true_def` makes it solvable by `word_bitwise`.

Reply:
Good suggestion, the following solves it without unfolding `true` and without the `word_bitwise`
```
lemma and_one_neq_zero:
  "a AND 1 ≠ 0 ⟹ a AND 1 = 1"
  by (clarsimp simp: and_one_eq)
  ```

Reply:
Funny thing, that's probably a `<-->` equivalence.
You can also make a version for `a AND 1 /= 1 ==> a AND 1 = 0`, and shove it into lib somewhere (unless we have something like that already, or Gerwin cooked one up in his recent Word_Lib enhancements)

Reply:
Word_Lib is a good place for `semiring_bit_operations` things. A lot of the things in Word_Lib are generalise to those now.

Reply:
Oh, and simpset: yes, this is looking good to be `[simp]`

Reply:
Yeah, I thought about it for a bit, thumbs up to simpset.
Potential hiccup: if you had `a AND 1 ≠ 0` and wanted to show `a ≠ 0`, it might be easier to show that than from `a AND 1 = 1` ... but how often does that happen?
For `a AND 1` this is less of a concern.